gpio/keypad/README.adoc

```diff
@@ -0,0 +1,120 @@
+= Keypad Example for Raspberry Pi Pico
+
+This document provides details about the 4x4 keypad example included in the larger repository of examples for Raspberry Pi Pico. This example demonstrates how to interface a 4x4 matrix keypad with the Pico and includes both simple and enhanced key detection modes.
+
+== Introduction
+
+This example shows how to use a 4x4 matrix keypad with the Raspberry Pi Pico. It has been enhanced to support detecting multiple key presses simultaneously, alongside the original single key press functionality.
+
+== Features
+
+* Simple key detection (one key press at a time)
+* Enhanced key detection (multiple keys pressed simultaneously)
+* Configurable maximum number of simultaneous key presses
+* Mode switching via macro definition
+
+== Hardware Requirements
+
+* Raspberry Pi Pico
+* 4x4 matrix keypad
+* Connecting wires
+
+== Wiring Instructions
+
+Follow these wiring instructions to connect the 4x4 matrix keypad to the Raspberry Pi Pico:
+
+[image2]
+image::pico_keypad_connection.png[Keypad Connection Diagram]
+
+=== Wiring Table
+
+The following table shows the GPIO pins used for the rows and columns of the keypad:
+
+[width="50%",cols="1,1,1",options="header"]
+|===
+| Keypad Row/Column | GPIO Pin | Pin Number
+
+| Row 0 | GP9  | 9
+| Row 1 | GP8  | 8
+| Row 2 | GP7  | 7
+| Row 3 | GP6  | 6
+
+| Column 0 | GP5  | 5
+| Column 1 | GP4  | 4
+| Column 2 | GP2  | 2
+| Column 3 | GP1  | 1
```

Comment:
You've written here that you're using GP1, GP2, GP4, GP5. But cross-referencing https://www.raspberrypi.com/documentation/microcontrollers/pico-series.html#pinout-and-design-files-2 with your wiring irmage, you seem to be using GP2, GP3, GP4, GP5, so now I'm super confused!